server/src/main/java/org/elasticsearch/indices/IndicesRequestCache.java

```diff
@@ -236,7 +248,8 @@ public boolean equals(Object o) {
             if (this == o) return true;
             if (o == null || getClass() != o.getClass()) return false;
             Key key = (Key) o;
-            if (Objects.equals(readerCacheKey, key.readerCacheKey) == false) return false;
+            if (mappingCacheKey.equals(key.mappingCacheKey) == false) return false;
```

Comment:
shouldn't we add mappingCacheKey to hashcode as well? Maybe we should expand IndicesRequestCacheTests so that it catches this?

Reply:
I'm not sure how I managed to miss that. I'll do it.

Reply:
78d5b334f2526363f58dc35fd343239bd3225c48